ert_shared/ensemble_evaluator/entity/function_step.py

```diff
@@ -30,8 +30,13 @@ async def _load(io_, transmitter):
         kwargs = {result[0]: result[1].data for result in results}
         function_output = func(**kwargs)
 
-        async def _transmit(io_, transmitter, data):
-            await transmitter.transmit_data(data)
+        async def _transmit(io_, transmitter: RecordTransmitter, data):
+            record: Record = (
+                BlobRecord(data=data)
+                if isinstance(data, bytes)
+                else NumericalRecord(data=data)
+            )
+            await transmitter.transmit_record(record)
             return (io_.get_name(), transmitter)
```

Comment:
Here (out of curiosity mainly), why the function `_transmit` returns `(name, transmitter)` as this is available beforehand?

Reply:
IIRC, it's because all of these transmits are done in one batch (see a couple of lines further down), so when all the transmitters are returned are returned in one bunch (the futures are done) you need to associate them again with their label.